crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1339,9 +1339,33 @@ impl<'db> TypeInferenceBuilder<'db> {
             .kind(self.db())
             .category(self.context.in_stub())
             .is_binding());
-        let use_def = self.index.use_def_map(binding.file_scope(self.db()));
-        let declarations = use_def.declarations_at_binding(binding);
+
+        let file_scope_id = binding.file_scope(self.db());
+        let symbol_table = self.index.symbol_table(file_scope_id);
+        let symbol_name = symbol_table.symbol(binding.symbol(self.db())).name();
+        let use_def = self.index.use_def_map(file_scope_id);
         let mut bound_ty = ty;
+
+        let is_global = self
+            .index
+            .globals_by_scope(file_scope_id)
+            .is_some_and(|globals| globals.contains(symbol_name));
+
+        let global_use_def_map = self.index.use_def_map(FileScopeId::global());
+        let declarations = if is_global {
```

Comment:
this gets rid of one of the failing seeds on the fuzzer CI job:

```suggestion
        let declarations = if is_global && file_scope_id != FileScopeId::global() {
```